tests/test_dagmc_stats.py

```diff
@@ -0,0 +1,142 @@
+# set the path to find the current installation of pyMOAB
+from pymoab import core, types
+from pymoab.rng import Range
+
+# import the new module that defines each of the functions
+import dagmc_stats
+import numpy as np
+import unittest
+
+test_input = "3vols.h5m"
+
+my_core = core.Core()
+my_core.load_file(test_input)
+
+root_set = my_core.get_root_set()
+entity_types = [types.MBVERTEX, types.MBTRI, types.MBENTITYSET]
+
+class TestDagmcStats(unittest.TestCase):
+
+
+    def test_get_tags(self):
+        """
+        Tests different aspects of the get_dagmc_tags function
+        """
+        dagmc_tags = dagmc_stats.get_dagmc_tags(my_core)
+        assert(len(dagmc_tags) == 3)
+        assert(dagmc_tags['category'])
+        assert(dagmc_tags['geom_dim'])
+        assert(dagmc_tags['global_id'])
+
+
+    def test_get_native_ranges(self):
+        """
+        Tests different aspects of the get_native_ranges function
+        """
+        native_ranges = dagmc_stats.get_native_ranges(my_core, root_set, entity_types)
+        vertex_range = my_core.get_entities_by_type(root_set, types.MBVERTEX)
+        assert(vertex_range == native_ranges[0])
+        triangle_range = my_core.get_entities_by_type(root_set, types.MBTRI)
+        assert(triangle_range == native_ranges[2])
+        entityset_range = my_core.get_entities_by_type(root_set, types.MBENTITYSET)
+        assert(entityset_range == native_ranges[11])
+
+
+    def test_get_entityset_ranges(self):
+        """
+        Tests different aspects of the get_entityset_ranges function
+        """
+        dagmc_tags = dagmc_stats.get_dagmc_tags(my_core)
+        entityset_ranges = dagmc_stats.get_entityset_ranges(my_core, root_set, dagmc_tags['geom_dim'])
+        node_range = my_core.get_entities_by_type_and_tag(root_set, types.MBENTITYSET, dagmc_tags['geom_dim'], [0])
+        assert(node_range == entityset_ranges['Nodes'])
+        curve_range = my_core.get_entities_by_type_and_tag(root_set, types.MBENTITYSET, dagmc_tags['geom_dim'], [1])
+        assert(curve_range == entityset_ranges['Curves'])
+        surface_range = my_core.get_entities_by_type_and_tag(root_set, types.MBENTITYSET, dagmc_tags['geom_dim'], [2])
+        assert(surface_range ==entityset_ranges['Surfaces'])
+        volume_range = my_core.get_entities_by_type_and_tag(root_set, types.MBENTITYSET, dagmc_tags['geom_dim'], [3])
+        assert(volume_range == entityset_ranges['Volumes'])
```

Comment:
In this PR I am not addressing the actual tests and whether or not they are good. I am just trying to get the framework setup so that the tests can actually be used at all. But that is a good comment about only doing one assertion per test (unless it can't be untangled).